app/routes/task_routes.py

```diff
@@ -1 +1,101 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from app.models.task import Task
+import os
+import requests
+from ..db import db
+from .route_utilities import validate_model, create_model, get_models_with_filters
+
+bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+# SEND SLACK NOTIFICATION
+def send_slack_notification(task_title):
+    slack_token = os.environ.get("SLACK_BOT_TOKEN")
+
+    url = "https://slack.com/api/chat.postMessage"
+
+    headers = {
+        "Authorization": f"Bearer {slack_token}",
+        "Content-Type": "application/json",
+    }
+
+    message_body = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {task_title}",
+    }
+
+    requests.post(url, headers=headers, json=message_body)
+
+
+# CREATE
+@bp.post("")
+def create_task():
+    request_body = request.get_json()
+
+    # Check for required fields before calling create_model
+    if "title" not in request_body or "description" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+
+    response = create_model(Task, request_body)
+    task_dict = response.get_json()
+    return make_response({"task": task_dict}, 201)
+
+
+# READ ALL
+@bp.get("")
+def get_all_tasks():
+
+    return get_models_with_filters(Task, request.args)
+
+
+# READ ONE
+@bp.get("/<task_id>")
+def get_one_task(task_id):
+    task = validate_model(Task, task_id)
+    return {"task": task.to_dict()}, 200
+
+
+# UPDATE
+@bp.put("/<task_id>")
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    db.session.commit()
+
+    return {"task": task.to_dict()}, 200
+
+# DELETE
+@bp.delete("/<task_id>")
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+    task_title = task.title
+    db.session.delete(task)
+    db.session.commit()
+
+    return {"details": f'Task {task_id} "{task_title}" successfully deleted'}, 200
```

Comment:
Same comment as before! Would there be a way to combine these delete methods into a utility function? And I know that it might seem a little weird knowing that our models might not all have the same attribute like "title". The nice part about building our own API is that we kind of get to design our models and our endpoints however we want! We could make sure every model has a title attribute!